domains/transfer/Transfer.API/API/Shared/Migrations/20231123093303_AddTransferAgreementProposalsAndDeleteConnections.cs

```diff
@@ -0,0 +1,85 @@
+using System;
+using Microsoft.EntityFrameworkCore.Migrations;
+
+#nullable disable
+
+namespace API.Shared.Migrations
+{
+    /// <inheritdoc />
+    public partial class AddTransferAgreementProposalsAndDeleteConnections : Migration
+    {
+        /// <inheritdoc />
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DropTable(
+                name: "ConnectionInvitations");
+
+            migrationBuilder.DropTable(
+                name: "Connections");
+
+            migrationBuilder.CreateTable(
+                name: "TransferAgreementProposals",
+                columns: table => new
+                {
+                    Id = table.Column<Guid>(type: "uuid", nullable: false),
+                    SenderCompanyId = table.Column<Guid>(type: "uuid", nullable: false),
+                    SenderCompanyTin = table.Column<string>(type: "text", nullable: false),
+                    SenderCompanyName = table.Column<string>(type: "text", nullable: false),
+                    CreatedAt = table.Column<DateTimeOffset>(type: "timestamp with time zone", nullable: false, defaultValueSql: "current_timestamp at time zone 'UTC'"),
+                    StartDate = table.Column<DateTimeOffset>(type: "timestamp with time zone", nullable: false),
+                    EndDate = table.Column<DateTimeOffset>(type: "timestamp with time zone", nullable: true),
+                    ReceiverCompanyTin = table.Column<string>(type: "text", nullable: true)
+                },
+                constraints: table =>
+                {
+                    table.PrimaryKey("PK_TransferAgreementProposals", x => x.Id);
+                });
+        }
```

Comment:
Would you get an exception if you attempted to delete a table, containing existing connections records? A migration test would determine that.